src/liboslexec/automata.cpp

```diff
@@ -240,10 +240,9 @@ NdfAutomata::lambdaClosure(IntSet &states)const
     while (discovered.size()) { // as long as there are new found states
         frontier.clear();
         // we do the same as in the above loop but with discovered instead of states
-        for (std::vector<int>::iterator i = discovered.begin(); i != discovered.end(); ++i) {
-            const State *state = m_states[*i];
-            std::pair <IntSet::const_iterator, IntSet::const_iterator> lr;
-            for (lr = state->getLambdaTransitions(); lr.first != lr.second; lr.first++) {
+        for (auto i : discovered) {
+            const State *state = m_states[i];
+            for (auto lr = state->getLambdaTransitions(); lr.first != lr.second; lr.first++) {
                 std::pair<IntSet::iterator, bool> rec = states.insert(*(lr.first));
```

Comment:
while your at it,
`std::pair<IntSet::iterator, bool> rec = states.insert(*(lr.first));`
could be changed to
`auto rec = states.insert(*(lr.first));`

Reply:
LGTM

Reply:
I have a second PR that fully modernizes. This set is *just* the things you did, which were directly needed to excise certain container names.